vkconfig/dlgcreateassociation.cpp

```diff
@@ -144,8 +149,7 @@ QTreeWidgetItem *dlgCreateAssociation::CreateApplicationItem(const Application &
     ui->treeWidget->addTopLevelItem(item);
 
     if (configurator.HasActiveOverrideOnApplicationListOnly()) {
-        item->setText(0, tr("    ") + application.executable_path);
-        QCheckBox *check_box = new QCheckBox("");
+        QCheckBox *check_box = new QCheckBox(application.executable_path);
```

Comment:
This looks to be related to the above... 

I don't know Qt well enough to understand the use of this structure, so I'll just ask: what happens if we repeatedly select and unselect whatever provides `HasActiveOverrideOnApplicationListOnly`?  Will we keep allocating new checkboxes each time?  Will the old checkboxes be cleaned up properly?

I'd be more comfortable with a static UI configuration, with elements that are made visible and invisible as needed, than I am with dynamically reconfiguring the UI repeatedly.

Reply:
These are only created when items are added to the list. We don't delete/re-add them when they are clicked/unclicked. Once added to the tree, they are owned and destroyed by the tree as well when it goes away. If the user deletes a row, that also destroys the tree. This is standard Qt design, all Widgets cleanup widgets that belong to them. Even if you replace the widget, the original widget get's destroyed by the set method for you so you don't have to keep a pointer around and do it yourself.